rotkehlchen/globaldb/manual_price_oracles.py

```diff
@@ -55,26 +60,42 @@ class ManualCurrentOracle(CurrentPriceOracleInterface):
 
     def __init__(self) -> None:
         super().__init__(oracle_name='manual current price oracle')
+        self.database: Optional['DBHandler'] = None
+
+    def set_database(self, database: 'DBHandler') -> None:
+        self.database = database
 
     def rate_limited_in_last(self, seconds: Optional[int] = None) -> bool:
         return False
 
-    def query_current_price(self, from_asset: Asset, to_asset: Asset) -> Price:
+    def query_current_price(
+            self,
+            from_asset: Asset,
+            to_asset: Asset,
+            match_main_currency: bool,
+    ) -> Tuple[Price, bool]:
         """
         Searches for a manually specified current price for the `from_asset`. If it finds
-        the price, converts it to a price in `to_asset` and returns.
+        the price and it is the main currency, returns as is and a boolean `True`.
+        Otherwise converts it to a price in `to_asset` and returns it + a boolean `False`.
         """
+        assert self.database is not None, 'Manual latest price oracle should have database set'
         manual_current_result = GlobalDBHandler().get_manual_current_price(
             asset=from_asset,
         )
         if manual_current_result is None:
-            return Price(ZERO)
-
+            return Price(ZERO), False
         current_to_asset, current_price = manual_current_result
+        if match_main_currency is True:
+            with self.database.conn.read_ctx() as cursor:
+                main_currency = self.database.get_setting(cursor=cursor, name='main_currency')
+                if main_currency != A_USD and current_to_asset == main_currency:  # noqa: E501
```

Comment:
Why do you need `if main_currency != A_USD` ?

Shouldn't you just follow the same thing if main currency is usd? It will save you extra calls later.

Reply:
I did it this way because I think that if the main currency is USD, no double conversion will happen. But yeah, it could make sense to be consistent and don't have this extra check

Reply:
yes it should be removed